meson.build

```diff
@@ -1320,73 +1341,83 @@ endif
 # Check for iconv support
 #
 
-with_libiconv = get_option('with-libiconv')
+enable_iconv = get_option('with-libiconv')
+iconv_path = get_option('with-libiconv-path')
+
+iconv = dependency('iconv', required: false)
+libiconv = cc.find_library('iconv', dirs: libsearch_dirs, required: false)
 
 libiconv_link_args = []
 
-if with_libiconv != ''
-    libiconv_link_args += ['-L' + with_libiconv / 'lib', '-liconv']
+if iconv_path != ''
+    libiconv_link_args += ['-L' + iconv_path / 'lib', '-liconv']
     if enable_rpath
-        libiconv_link_args += ['-R' + with_libiconv / 'lib']
+        libiconv_link_args += ['-R' + iconv_path / 'lib']
     endif
     iconv = declare_dependency(
         link_args: libiconv_link_args,
         include_directories: include_directories(with_libiconv / 'include'),
     )
-else
-    iconv = dependency('iconv', required: false)
 endif
 
-libiconv = cc.find_library('iconv', dirs: libsearch_dirs, required: false)
-
-if iconv.found() or libiconv.found()
-    if cc.compiles(
-        '''
+if not enable_iconv
+    have_iconv = false
+else
+    have_iconv = (
+        (iconv.found() or libiconv.found())
+        and cc.compiles(
+            '''
```

Comment:
Uneven indentation with the closing `'''` again – but maybe this is deliberate?

Reply:
As above